SAK-29810 New SakaiScript Site related webservices

ddelblanco opened the pull request:
changeSitePublishStatus (passing the siteid and a boolean to select the
state)
checkForMemberInSite (passing siteid and eid return boolean)
getAvailableRoles (get the roles passing the authz)
getExistingFunctions (get the functions of a role passing the roleid
and the authz)
getSiteDefaultJoinerRole (know the default sitejoiner role apssing the
siteid)

steveswinsburg commented on webservices/cxf/src/java/org/sakaiproject/webservices/SakaiScript.java:
```diff
@@ -4257,4 +4257,148 @@ private void setToolProperties(ToolConfiguration tool, String propList) {
+            if (site != null) {
+                return site.getJoinerRole();
+            } else {
+                return "failure";
```
This should probably not return a String as the error when the OK response is also a String. Also it will look like an error if the joiner role is actually 'failure'. Throw an exception or log it and return nothing IMO.

ddelblanco replied:
Totally agree. I'll change it and resend. Thanks Steve

ottenhoff commented on webservices/cxf/src/java/org/sakaiproject/webservices/SakaiScript.java:
```diff
@@ -4257,4 +4257,296 @@ private void setToolProperties(ToolConfiguration tool, String propList) {
+
+        Set roles;
+        Iterator iRoles;
+        boolean createRole = false;
```
This createRole variable doesn't seem to be used

ddelblanco replied:
Removed, and the next line too (role) was not used neither.

ottenhoff commented on webservices/cxf/src/java/org/sakaiproject/webservices/SakaiScript.java:
```diff
@@ -4257,4 +4257,296 @@ private void setToolProperties(ToolConfiguration tool, String propList) {
+    @Path("/changeSitePublishStatus")
+    @Produces("text/plain")
+    @GET
+    public String changeSitePublishStatus(
```
This looks duplicated

ddelblanco replied:
Yeah... seems that intelliJ is not so intelligent as expected. For some reason is duplicated... maybe git...

ddelblanco commented:
I've rebased and reviewed this code in only one commit.
Added the superuser security to the ones that are not controlled by permissions. 
And tested again.

ottenhoff commented:
Can you apply the isSuperuser check to changeSitePublishStatus ?

Then good to merge

ddelblanco commented:
I've tested it with a "normal" user and only if the user has the site.publish permission it works, if not it shows a non permission exception (as most of the other ones that change site things). I can add the superuser check, it  takes me only a minute, but do it really needs it?

ottenhoff commented:
Okay, as long as it denies permission, it's good to go... feel free to merge when ready

ddelblanco commented:
Thanks!  I can't merge myself. Can you do it, please?
